### PR TITLE
Auto reconnection when channel state changed to idle

### LINKED ISSUE #1844: [FEATURE]: Try auto reconnect when connection state changes to idle
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

Part of #1843

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by haorenfsa**
Fix #1844
/cc @XuanYang-cn

### CODE REVIEW

File: pymilvus/orm/connections.py [hunk 1]
```diff
@@ -293,13 +344,19 @@ def connect(
...
         def connect_milvus(**kwargs):
             gh = GrpcHandler(**kwargs)
 
             t = kwargs.get("timeout")
             timeout = t if isinstance(t, (int, float)) else Config.MILVUS_CONN_TIMEOUT
 
             gh._wait_for_channel_ready(timeout=timeout)
+            gh.register_state_change_callback(
+                ReconnectHandler(self, alias, kwargs_copy).reconnect_on_idle
+            )
```
Comment (XuanYang-cn) [inline pymilvus/orm/connections.py:359]: please add a flag to control whether to add the callback or not, for instance, reset_connection=True.

Do not set it as default bahaviour to aviod unknown bugs. After all It's a temporay solution.

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

File: pymilvus/orm/connections.py [hunk 2]
```diff
@@ -293,13 +344,19 @@ def connect(
             >>> connections.connect("test", host="localhost", port="19530")
         """
 
+        # kwargs_copy is used for auto reconnect
+        kwargs_copy = copy.deepcopy(kwargs)
```
[the remark this answers is not in the document]
Comment (haorenfsa) [reply pymilvus/orm/connections.py:350]: updated

File: pymilvus/orm/connections.py (hunk 1 above)
Comment (haorenfsa) [reply pymilvus/orm/connections.py]: updated

**Pull request merged**